cmake/OpenCVFindLibsGrfmt.cmake

```diff
@@ -222,38 +222,52 @@ if(WITH_JASPER AND NOT HAVE_OPENJPEG)
 endif()
 
 # --- libpng (optional, should be searched after zlib) ---
-if(WITH_PNG)
-  if(BUILD_PNG)
-    ocv_clear_vars(PNG_FOUND)
+if(WITH_PNG OR WITH_SPNG)
+  if(WITH_SPNG)
+    set(SPNG_LIBRARY libspng CACHE INTERNAL "")
+    set(SPNG_LIBRARIES ${SPNG_LIBRARY})
+    add_subdirectory("${OpenCV_SOURCE_DIR}/3rdparty/libspng")
+    set(SPNG_INCLUDE_DIR "${${SPNG_LIBRARY}_SOURCE_DIR}" CACHE INTERNAL "")
+    set(SPNG_DEFINITIONS "")
+    ocv_parse_header("${SPNG_INCLUDE_DIR}/spng.h" SPNG_VERSION_LINES SPNG_VERSION_MAJOR SPNG_VERSION_MINOR SPNG_VERSION_PATCH)
+
+    set(HAVE_SPNG YES)
+    set(SPNG_VERSION "${SPNG_VERSION_MAJOR}.${SPNG_VERSION_MINOR}.${SPNG_VERSION_PATCH}")
+    message(STATUS "imgcodecs: PNG codec will use SPNG, version: ${SPNG_VERSION} ")
   else()
-    ocv_clear_internal_cache_vars(PNG_LIBRARY PNG_INCLUDE_DIR)
-    include(FindPNG)
-    if(PNG_FOUND)
-      include(CheckIncludeFile)
-      check_include_file("${PNG_PNG_INCLUDE_DIR}/libpng/png.h" HAVE_LIBPNG_PNG_H)
-      if(HAVE_LIBPNG_PNG_H)
-        ocv_parse_header("${PNG_PNG_INCLUDE_DIR}/libpng/png.h" PNG_VERSION_LINES PNG_LIBPNG_VER_MAJOR PNG_LIBPNG_VER_MINOR PNG_LIBPNG_VER_RELEASE)
-      else()
-        ocv_parse_header("${PNG_PNG_INCLUDE_DIR}/png.h" PNG_VERSION_LINES PNG_LIBPNG_VER_MAJOR PNG_LIBPNG_VER_MINOR PNG_LIBPNG_VER_RELEASE)
+    if(BUILD_PNG)
+      ocv_clear_vars(PNG_FOUND)
+    else()
+      ocv_clear_internal_cache_vars(PNG_LIBRARY PNG_INCLUDE_DIR)
+      include(FindPNG)
+      if(PNG_FOUND)
```

Comment:
Please minimize amount of change code:

```
if(WITH_SPNG)
   ... new code
endif()
if(NOT HAVE_SPNG AND WITH_PNG)
   ... existed untouched code ...
endif()
```
```

Reply:
@alalek, instead of putting generic comments, could you please put the snippet what you have in mind? I personally cannot reverse engineer it from your comment